sensor/kubernetes/listener/resources/registrymirrorsets.go

```diff
@@ -0,0 +1,47 @@
+package resources
+
+import (
+	configV1 "github.com/openshift/api/config/v1"
+	operatorV1Alpha1 "github.com/openshift/api/operator/v1alpha1"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/sensor/kubernetes/eventpipeline/component"
+)
+
+type registryMirrorDispatcher struct {
+}
+
+func newRegistryMirrorDispatcher() *registryMirrorDispatcher {
+	return &registryMirrorDispatcher{}
+}
+
+// ProcessEvent processes registry mirroring related resource events and returns the sensor events to emit in response.
+func (r *registryMirrorDispatcher) ProcessEvent(obj, _ interface{}, action central.ResourceAction) *component.ResourceEvent {
+	switch v := obj.(type) {
+	case *operatorV1Alpha1.ImageContentSourcePolicy:
+		return r.handleImageContentSourcePolicy(v, action)
+	case *configV1.ImageDigestMirrorSet:
+		return r.handleImageDigestMirrorSet(v, action)
+	case *configV1.ImageTagMirrorSet:
+		return r.handleImageTagMirrorSet(v, action)
+	}
+
+	return nil
```

Comment:
maybe default to a message saying the type is not supported?

Reply:
Don't believe that would ever happen 'naturally' because only the resources we've pointed the informers at in `sensor/kubernetes/listener/resource_event_handler.go` should go to this handler.  

But just in case its done accidentally in the future -  I'll add a message.

Also looking at this closer the `return nil` should be problem (causing a panic) - I see the return nil in another handler and seems to be working however, will investigate further.

Reply:
False alarm - all `dispatchers` by are wrapped in a `MetricDispatcher` in `sensor/kubernetes/listener/resources/dispatcher.go` which handles the nil return.